File: src/main/scala/org/lolhens/minechanics/common/util/UnlocalizedNameUtil.scala
package org.lolhens.minechanics.common.util

import net.minecraft.block.Block

object UnlocalizedNameUtil {
  def getBlockByUnlocalizedName(unlocalizedNameArg: String): Block = {
    val unlocalizedName = unlocalizedNameArg.toLowerCase()
    val iterator = Block.blockRegistry.iterator()
    while (iterator.hasNext()) {
      val obj = iterator.next()
      if (obj.isInstanceOf[Block] && obj.asInstanceOf[Block].getUnlocalizedName().toLowerCase() == unlocalizedName)
        return obj.asInstanceOf[Block]
    }
    null
  }
}

File: src/main/java/org/lolhens/minechanics/core/util/UnlocalizedNameUtil.java
package org.lolhens.minechanics.core.util;

import java.util.Iterator;

import net.minecraft.block.Block;

public class UnlocalizedNameUtil {
    public static Block getBlockByUnlocalizedName(String unlocalizedName) {
        unlocalizedName = unlocalizedName.toLowerCase();
        Iterator<?> iterator = Block.blockRegistry.iterator();
        while (iterator.hasNext()) {
            Object obj = iterator.next();
            if (obj instanceof Block && ((Block) obj).getUnlocalizedName().toLowerCase().equals(unlocalizedName))
                return (Block) obj;
        }
        return null;
    }
}
